this patch depends on the Shlwapi.lib changes from better_console.patch!

diff --git src/fpsgame/client.cpp src/fpsgame/client.cpp
index 6cf20f3..5a10e62 100644
--- src/fpsgame/client.cpp
+++ src/fpsgame/client.cpp
@@ -1,6 +1,8 @@
 #include "game.h"
 #include "weaponstats.h"
-
+#ifdef WIN32
+#include <shlwapi.h> // for StrStrIA (= strcasestr)
+#endif
 namespace game
 {
     VARP(minradarscale, 0, 384, 10000);
@@ -380,12 +382,31 @@ namespace game
             fpsent *o = players[i];
             if(!strcmp(arg, o->name)) return o->clientnum;
         }
-        // nothing found, try case insensitive
+        // try case insensitive
         loopv(players)
         {
             fpsent *o = players[i];
             if(!strcasecmp(arg, o->name)) return o->clientnum;
         }
+        if(strlen(arg)>2)
+        {
+            // try case insensitive prefix
+            loopv(players)
+            {
+                fpsent *o = players[i];
+                if(!strncasecmp(arg, o->name, strlen(arg))) return o->clientnum;
+            }
+            // try case insensitive substring
+            loopv(players)
+            {
+                fpsent *o = players[i];
+#ifdef WIN32
+                if(StrStrIA(o->name, arg)) return o->clientnum;
+#else
+                if(strcasestr(o->name, arg)) return o->clientnum;
+#endif
+            }
+        }
         return -1;
     }
     ICOMMAND(getclientnum, "s", (char *name), intret(name[0] ? parseplayer(name) : player1->clientnum));
@@ -504,10 +525,10 @@ namespace game
         }
     });
 
-    void setteam(const char *arg1, const char *arg2)
+    void setteam(const char *playername, const char *teamname)
     {
-        int i = parseplayer(arg1);
-        if(i>=0) addmsg(N_SETTEAM, "ris", i, arg2);
+        int i = parseplayer(playername);
+        if(i>=0) addmsg(N_SETTEAM, "ris", i, teamname);
     }
     COMMAND(setteam, "ss");
 
